packages/eslint-plugin/lib/rules/use-baseline.js

```diff
@@ -0,0 +1,226 @@
+    }
+
+    /**
+     * @param {string} encoded
+     * @returns {[number, number]}
+     */
+    function decodeStatus(encoded) {
+      const [status, year = NaN] = encoded
+        .split(":")
+        .map((part) => Number(part));
+      return [status, year];
+    }
+
+    /**
+     * @param {string} encoded
+     * @returns {boolean}
+     */
+    function isSupported(encoded) {
+      const [status, year = NaN] = decodeStatus(encoded);
+      if (baseYear) {
+        return year <= baseYear;
+      }
+      return status >= baseStatus;
+    }
+
+    /**
+     * @param {string} element
+     * @returns {boolean}
+     */
+    function isSupportedElement(element) {
+      const elementEncoded = elements.get(element);
+      if (!elementEncoded) {
+        return true;
+      }
+      return isSupported(elementEncoded);
+    }
+
+    /**
+     * @param {string[]} parts
+     * @returns {string}
+     */
+    function toStatusKey(...parts) {
+      return parts.map((part) => part.toLowerCase().trim()).join(".");
+    }
+
+    /**
+     * @param {string} element
+     * @param {string} key
+     * @returns {boolean}
+     */
+    function isSupportedAttributeKey(element, key) {
+      const elementStatus = elements.get(toStatusKey(element, key));
+      if (elementStatus) {
+        return isSupported(elementStatus);
+      }
+      const globalAttrStatus = globalAttributes.get(toStatusKey(key));
+      if (!globalAttrStatus) {
+        return true;
+      }
+      return isSupported(globalAttrStatus);
+    }
+
+    /**
+     * @param {string} element
+     * @param {string} key
+     * @param {string} value
+     * @returns {boolean}
+     */
+    function isSupportedAttributeKeyValue(element, key, value) {
+      const elementStatus = elements.get(toStatusKey(element, key, value));
+
+      if (elementStatus) {
+        return isSupported(elementStatus);
+      }
+      const globalAttrStatus = globalAttributes.get(toStatusKey(key, value));
+      if (!globalAttrStatus) {
+        return true;
+      }
+      return isSupported(globalAttrStatus);
+    }
+
+    /**
+     * @param {Tag | ScriptTag | StyleTag} node
+     * @param {string} elementName
+     * @param {Attribute[]} attributes
+     */
+    function check(node, elementName, attributes) {
+      if (isCustomElement(elementName)) {
+        return;
+      }
+
+      if (!isSupportedElement(elementName)) {
+        context.report({
+          node: node.openStart,
+          messageId: MESSAGE_IDS.NOT_BASELINE_ELEMENT,
+          data: {
+            element: elementName,
+            availability,
+          },
+        });
+      }
+      attributes.forEach((attribute) => {
+        if (!isSupportedAttributeKey(elementName, attribute.key.value)) {
+          context.report({
+            node: attribute.key,
+            messageId: MESSAGE_IDS.NOT_BASELINE_ATTRIBUTE,
+            data: {
+              element: elementName,
+              attr: attribute.key.value,
```

Comment:
It might be useful to distinguish between global and element-specific attributes. If it's element-specific, we could show it as `element[attribute]` for additional context.

Reply:
@rviscomi Thanks! 
I changed the message to separate the element/global attribute context and make it a little easier to understand.
https://github.com/yeonjuan/html-eslint/pull/311/commits/c8182e103e62f4a486385ae4cc417f0290e55f01